tests/suites/os/tests/device-specific-tests/beaglebone-black/index.js

```diff
@@ -58,5 +72,28 @@ module.exports = {
 			false,
 			'HDMI is now disabled (overlay not present in /proc/device-tree/chosen/overlays/)',
 		);
+
+		// Revert the uEnv.txt changes and reboot the DUT so when we finish the test we have the initial setup
+		test.comment('Revert changes to uEnv.txt and reboot ...');
+		await this.context
+			.get()
+			.worker.executeCommandInHostOS(
+				`head -n -2 /mnt/boot/uEnv.txt > /mnt/boot/temp_uEnv.txt && mv /mnt/boot/temp_uEnv.txt /mnt/boot/uEnv.txt`,
+				this.context.get().link,
+			};
+		await this.context.get().worker.rebootDut(this.context.get().link);
+
```

Comment:
This device does not boot very fast and I'm afraid the final test may fail in some circumstances. What if we added these lines below right here, to check every 10 seconds for up to 5 minutes (or some other timeout) and ensure we proceed with the final test only after the device is up:

                         ```
       test.comment(`Waiting for device to finish booting`);
                                await this.context.get().utils.waitUntil(async () => {
                                        return (
                                                (await this.context
                                                        .get()
                                                        .worker.executeCommandInHostOS(
                                                                `echo booted`,
                                                                this.context.get().link,
                                                        )) === 'booted'
                                        );
                                },
                                false,
                                30 /* 30 attempts */
                                10000 /* every 10 seconds */
                                );
```

Reply:
@rcooke-warwick I believe rebootDut waits for 100 seconds now? https://github.com/balena-os/leviathan/blob/master/core/lib/common/worker.js#L348 Could we make it universal for all boards and increase this interval to a few minutes? (3-4 minutes maybe) This would be beneficial for other boards which are slower to boot

Reply:
@floion @acostach - the current timeout is most likely enough. It waits for longer than 100 seconds, it looks like it actually waits 100*10 = 1000 seconds ~ 16mins! @vipulgupta2048 maybe we can adjust this in leviathan, or even make the timeout configurable with an arguments with adjusted defaults.

Reply:
@rcooke-warwick I see { interval: 10000, tries: 10 }, isn't this 10 * 1000ms  * 10 (retries) ?

Reply:
@acostach sorry you're right, its 10 seconds * 10 tries, Ignore what I said earlier ;/ In that case we should raise an issue on the leviathan repo I think, and discuss it there